maple2latex/src/translator.py

```diff
@@ -68,6 +72,126 @@ def __init__(self, inp):
             self.general = [self.fields["even"], self.fields["odd"]]
 
 
+class LatexEquation(object):
+    def __init__(self, label, equation, metadata):
+        self.label = label
+        self.equation = equation
+        self.metadata = metadata
+
+    @classmethod
+    def from_maple(cls, eq):
+        # modify fields
+        eq.lhs = translate(eq.lhs)
+        eq.factor = translate(eq.fields["factor"])
+        eq.front = translate(eq.fields["front"])
+
+        equation = eq.lhs + "\n  = "
+
+        if eq.factor == "1":
+            eq.factor = ""
+
+        metadata = dict()
+
+        # translates the Maple information (with spacing)
+        if eq.eq_type == "series":
+            eq.general = translate(eq.general[0])
+
+            if eq.factor != "":
+                equation += eq.factor + " "
+
+            elif eq.front != "":
+                equation += eq.front + "+"
+
+            equation += "\\sum_{k=0}^\\infty "
+
+            if eq.category == "power series":
+                equation += eq.general
+            elif eq.category == "asymptotic series":  # make sure to fix asymptotic series
+                equation += "(" + eq.general + ")"
+
+        elif eq.eq_type == "contfrac":
+            forms = list()
+
+            # substitution handling
+            if len(parse_brackets(eq.general[0])) > 1:
```

Comment:
`parse_brackets(eq.general[0])` occurs multiple times in this vicinity, but instead of being stored, the function is just called every time the value is needed.